homeassistant/auth/__init__.py

```diff
@@ -341,8 +280,52 @@ def async_auth_providers(self):
 
     async def async_remove_user(self, user):
         """Remove a user."""
+        tasks = [
+            self.async_remove_credentials(credentials)
+            for credentials in user.credentials
+        ]
+
+        if tasks:
+            await asyncio.wait(tasks)
         await self._store.async_remove_user(user)
 
+    async def async_remove_credentials(self, credentials):
+        """Remove credentials."""
+        provider = self._async_get_auth_provider(credentials)
+
+        if (provider is not None and
+                hasattr(provider, 'async_will_remove_credentials')):
+            await provider.async_will_remove_credentials(credentials)
+
+        await self._store.async_remove_credentials(credentials)
+
+    async def async_enable_user_mfa(self, user, mfa_module_id, **kwargs):
```

Comment:
We shouldn't use `**kwargs`, if the module expects certain data, we should just store that in a dictionary and pass that on.